### ISSUE TITLE
No version control on the FAQ system

### ISSUE THREAD

**Issue by me-and** [dreamwidth/dw-free#1455]
Apparently there is no version control on the FAQ system. This is the sort of thing that makes me sad, and I'd quite like to fix it.

I expect this isn't trivial, but I can't imagine it being *too* horrific...

**Comment by azurelunatic** [comment]
Oh god why do you *say* things like that?!

**Comment by me-and** [comment]
I should probably claim this...

**Comment by me-and** [comment]
Right, I have the outline of a solution, which is basically to implement [this Stack Exchange solution](http://stackoverflow.com/a/323109). Short version goes:

* Add a new table `faqhist`, which has all the fields the `faq` table has, plus `revision`, where the `revision` field is an additional index field. (Timestamps are also useful but they're already present in that table.)
* Every time an FAQ entry is created or edited, add it to the `faqhist` table as well as the `faq` table, incrementing the `revision` from the previous entry with the matching index fields.
* Bob's your uncle/aunt/appropriate gender-neutral relative.

**Comment by rahaeli** [comment]
I leave this quote from the qdb as being relevant: http://qdb.dreamwidth.net/dw/72

**Comment by zarhooie** [comment]
Kab and Adam have both been prepped on the signs of being possessed by an Elder God, or whatever else may be shambling in the deep, dark depths of the FAQ system.

**Comment by azurelunatic** [comment]
So when the badgers go all wrong and someone manages to save some FAQ as
blank, what's the rollback process?

**Comment by me-and** [comment]
The rollback process will just be pulling the data out of the database manually and copying it into place. Once the information is being recorded, though, it'll be possible to add a web interface into the table to view old entries, restore them, view diffs, etc.

**Comment by me-and** [comment]
There, see, not too horrific at all!

**Comment by rahaeli** [comment]
I really, really misread 'faqhist'. ;)

**Comment by zorkian** [comment]
Right?!

--
Mark Smith
mark@qq.is
 
 
On Thu, Aug 6, 2015, at 05:56 PM, Denise Paolucci wrote:
> I really, really misread 'faqhist'. ;)


> —
> Reply to this email directly or view it on GitHub[1]. 


>  


Links:

  1. https://github.com/dreamwidth/dw-free/issues/1455#issuecomment-128551762

**Comment by me-and** [comment]
...it took me far longer than I'd like to admit to work out what you could have possibly misread it as ><

Want me to change it?

**Comment by rahaeli** [comment]
Nah, Adam, I just have a dirty mind! :) It's a perfectly logical name for the table, it just happens to be a typo I make frequently when I'm aiming in the other direction...
